torch/_dynamo/variables/builtin.py

```diff
@@ -514,17 +535,20 @@ def call_function(
                         *proxy_args_kwargs(args, kwargs),
                     )
 
-                    return wrap_fx_proxy_cls(
+                    ret = wrap_fx_proxy_cls(
                         variables.NumpyNdarrayVariable, tx, proxy, **options
                     )
+                    if self.fn in self._self_assigning_ops():
+                        tx.replace_all(args[0], ret)
```

Comment:
Doesn't this require `.mutable_local` being set?

Reply:
Yes, it does, and will assert. We need to do a little more work here to properly make sure our mutable_local story is lined up for replace_all.

Reply:
Yes, it does indeed. fixed now

Reply:
We will copy the mutable localin replace_all if it is an attribute mutation.